Packages/com.unity.inputsystem/InputSystem/SourceGenerators/Source~/SourceGenerators/GlobalInputActionsSourceGenerator.cs

```diff
@@ -0,0 +1,372 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Unity.InputSystem.SourceGenerators
+{
+    [Generator]
+    public class GlobalInputActionsSourceGenerator : ISourceGenerator
+    {
+        //@TODO: This needs a JSON file
```

Comment:
Why do we have this TODO. The code below seems to handle extracting the incoming data out into a JSON file so shouldn't we just make this a statement.

Reply:
Hmm the more critical aspect seems to be this should not reference the template. It needs to access the InputManager.asset

Reply:
Noted in PR description against [ISX-1530](https://jira.unity3d.com/browse/ISX-1530) and now resolved.